File: Baekjoon/BaaarkingDog/0x03_array/1406.cpp
// 1406: 에디터

#include <iostream>
#include <string>
#include <stack>
using namespace std;

int main() {
  string string1;
  stack<char> st_left;
  stack<char> st_right;
  int M;
  cin >> string1 >> M;

  for (char c : string1) st_left.push(c);

  while (M--) {
    char cmd;
    
    cin >> cmd;
    if (cmd == 'P') {
      char key;
      cin >> key;
      st_left.push(key);
    } else if (cmd == 'L' && st_left.size() > 0) {
      st_right.push(st_left.top());
      st_left.pop();
    } else if (cmd == 'D' && st_right.size() > 0) {
      st_left.push(st_right.top());
      st_right.pop();
    } else if (cmd == 'B' && st_left.size() > 0) {
      st_left.pop();
    }
  }

  while (!st_left.empty()) {
    st_right.push(st_left.top());
    st_left.pop();
  }

  while (!st_right.empty()) {
    cout << st_right.top();
    st_right.pop();
  }

  return 0;
}

File: Baekjoon/BaaarkingDog/0x03_array/1406.py
# 1406: 에디터

st_left = []
st_right = []

for c in input(): st_left.append(c)

M = int(input())

for _ in range(M):
  cmd = input()

  if (cmd == 'L' and len(st_left) > 0):
    st_right.append(st_left.pop())
  elif (cmd == 'D' and len(st_right) > 0):
    st_left.append(st_right.pop())
  elif (cmd == 'B' and len(st_left) > 0):
    st_left.pop()
  elif len(cmd.split()) == 2:
    _, key = cmd.split()
    st_left.append(key)

while (len(st_left) != 0): st_right.append(st_left.pop())

while(len(st_right) != 0): print(st_right.pop(), end="")
